### ISSUE TITLE
Implement CC design intent feedback

### ISSUE THREAD

**Issue by jilladams** [department-of-veterans-affairs/va.gov-cms#14533]
Will resolve: 
https://github.com/department-of-veterans-affairs/va.gov-team/issues/62212
https://github.com/department-of-veterans-affairs/va.gov-team/issues/62175
https://github.com/department-of-veterans-affairs/va.gov-team/issues/62285

Michelle signoff on scope of Shoulds / Considers we want to address: https://github.com/department-of-veterans-affairs/va.gov-cms/issues/14087#issuecomment-1650582970

## ACs: 
- [x] Replace modal with drawer for use with multiple filters
- [x] Review [icons library](https://design.va.gov/foundation/icons) for suitable option to use for provider core training complete,
- [x] Investigate [US system segmented button group](https://designsystem.digital.gov/components/button-group/) as possible replacement for tabs and check on tech implementation
- [x] Explore other placements for the button that opens the filter modal/dropdown
- [x] Internally review & sign off on updates @mmiddaugh

**Comment by jilladams** [comment]
Noting: this was full notes on the feedback from those 3 tickets that led to this consolidated list: 

https://github.com/department-of-veterans-affairs/va.gov-team/issues/62212

- **Should**: Like others in the meeting, I prefer the drop-down accordion button to the modal design you mocked up. But I agree with @briandeconinck and am wary of the placement (on Desktop) of it off to the far right / share concerns of it getting lost. If you have good rationale to keep it there, definitely include findability tasks for Veterans around this component in user research.
  - Dropdown is for 1 filter. 
  - Need to use drawer when we have multiple filters (and that will replace modal) -- we should implement this before Midpoint review.
- **Consider**: is there any reason users may want to **sort** their search results on some criteria other than closest distance?
    - No strong opinion
- **Consider**: I love the thought behind the "types of care" search index that leverages typeahead search and additional keywords added to those care types. Have you considered also linking to the full list of the 110+ types of care so that users can look at that in full (vs. only being able to keyword search against it)?
    - Have considered & rejected that.

https://github.com/department-of-veterans-affairs/va.gov-team/issues/62175

- Consider: We are trying to align icons across VA. Before you create a new icon for provider core training complete, look at our [icons library](https://design.va.gov/foundation/icons) and see if one fits your need.
    - Yes, will investigate/implement before Midpoint
- Consider: Matt Dingee: We don’t love these tabs and have been trying to get rid of them. I think at some point you could replace that with the [US system segmented button group](https://designsystem.digital.gov/components/button-group/).
    - Yes, would like to investigate/implement before Midpoint. Need to check tech implementation.
- Consider: Matt Dingee: I don’t love the filters in the modal. Consider using the drawer. [GI Bill](https://app.zenhub.com/workspaces/collaboration-cycle-reviews-61d89e9e5f4c120011a0ad37/issues/gh/department-of-veterans-affairs/va.gov-team/44187) was working on a drawer option as well. Modals typically are there to stop an action.
    - Same, will use drawer for multiples

https://github.com/department-of-veterans-affairs/va.gov-team/issues/62285
- Should: I don't have any specific objection to using a modal for filters, except that modals are a big enough task interruption that if they can be avoided it would be nice to avoid them. But otherwise I defer to design feedback on consistency with other tools.
    - Handled elsewhere
- Should: Explore other placements for the button that opens the filter modal/dropdown -- or, if possible, validate this placement via usability testing. My main concern is low vision users who use screen magnifiers (as opposed to using browser zoom), who might miss the button as they focus on the left-hand column listing results.
    - No strong feelings
- Potential accessibility improvements: To the extent that you are able to prioritize accessibility tech debt, I would love to see:
    - Better focus management when a map marker is clicked. Currently when you click a map marker the results list scrolls to the corresponding entry and adds the active class to highlight it. But although the highlight border looks the same as the keyboard focus indicator, it doesn't actually move keyboard focus --- so if I press [tab] I'll be taken somewhere unexpected.
    - Map markers should receive keyboard focus and be keyboard-interactive ([enter] key to activate marker and move focus to the corresponding entry in the results list).
    - Map markers should have a unique accessible name. If they're keyboard accessible, something useful should be announced to screen readers. Map accessibility is a huge pain and it's difficult to create a fully accessible experience for screen reader users. At minimum, screen reader users should know they've tabbed to a map marker that's interactive, and be able to distinguish it from other markers. Better yet would be if there's some way to communicate basic geographic information too --- maybe an address or just a city. Honestly no idea what's possible here, just giving you my wishlist!
    - Another wishlist item: keyboard-accessible controls for moving the map north/south/east/west.
    - If map markers can receive keyboard focus, then provide keyboard users a skip link to jump them past the map and to the pagination (so they don't have to tab through 10 markers plus map controls if they don't want to).
        - MAKE A TICKET: Mostly FE implementation notes, that we need to keep track of in a ticket. (Jill to file.)

**Comment by thejordanwood** [comment]
1. The modal has been replaced with a drawer, similar to what the [GI bill](https://accenturefederal.invisionapp.com/freehand/Comparison-Tool-Enhancements---Design-Intent-CBjMzoSgB?zoomToItems=Y2w1dHJkYTFkMDFkMzN5ODJzZXgwbTQ3dQ%3D%3D) is using. Not sure if we want to completely align with their design. It looks like their desktop design is using a modal. Updated design can be found on the **Filters** page of the [Sketch file](https://sketch.com/s/d5e5c924-e8c0-47af-b3e4-994c335d6f31). I can update everywhere once this is approved. 
2. The icon I selected is already in the Profile section of the icon library. Should I select a different one?
3. I updated the tabs to a segmented button. This can be found on the **Mobile Map** page of the Sketch file. 
4. I would like to discuss the best placement for the filter button with the group. My earlier designs have placed this next to the Search button and another design had it on the left side instead of the right. We placed this to the right because it seemed like a logical place for screen readers to pick it up. They reading order would be: results fields > "Showing 1-10 results..." > filter these results. I agree that it gets a bit lost visually and am willing to keep brainstorming this. Just wanted to explain the reasoning behind the current placement.

**Comment by aklausmeier** [comment]
Responses to @thejordanwood 

1. Drawer: Great that you considered all viewport sizes. Agree with fill screen vs. bottom up functionality as it will allow this UI pattern to be scalable for additional filters if/when needed and for scalability into other areas where filters are used on PW. 
2. Icon: Based on design intent, you are good with the icon you chose because it's in the current VA design system so now that we know we can point that out in Midpoint, however just to have in your back pocket...
After looking at the USWDS v3, there doesn't seem to be an equivalent to the profile with check icon. So what might happen when icons on VA design system are updated to USWDS?  Because the one you chose is labeled as "payment history", the newer icon that is chosen might not make sense. A possible alternative IF provider training is always "medical" is to use the Healthcare icon in [Hubs](https://design.va.gov/foundation/icons#hub) since there is a near match in the USWDS icons.
3. Segmented button is great. You've left enough space between filter button for touch target area.
4. Seems the only concern is on desktop. Consider trying the button on left side directly under the "Showing 1–10 of 2234 results for "Community providers (in VA’s network)", "Chiropractor" near "New York, New York"
This would keep things in same reading order but move button to left align to address accessibility's concern in design intent.

**Comment by thejordanwood** [comment]
**End of sprint update:** 

These changes are made and I'm waiting to review them with the team in UX refinement (potentially on 8/17).

**Comment by jilladams** [comment]
@thejordanwood can close now?

**Comment by thejordanwood** [comment]
@jilladams Yes, although we aren't going to Midpoint Review at this time and that's the last AC.

**Comment by jilladams** [comment]
ah right! @xiongjaneg @thejordanwood I'll leave it to you 2 to decide how we wanna track that dangler (its own ticket? leave this one open? etc)

**Comment by xiongjaneg** [comment]
@thejordanwood Does this not need to go to Midpoint Review or just not yet? If not until later, I'm inclined to move that AC into a separate ticket for a separate sprint and close this one.

**Comment by thejordanwood** [comment]
@xiongjaneg I believe we decided not to go to Midpoint until we're clear on what is going to be MVP for this. There's been some discussion about not including search filters in MVP. 

We do need to open a content review ticket for this so that team can look this over. 

Feel free to move whatever you need to into another ticket! I'm fine with closing this one because the design intent feedback has been implemented.

**Comment by thejordanwood** [comment]
@xiongjaneg Just a couple of thoughts I want to leave here:

1. We got feedback from the design intent meeting that we should explore including a list of all health services offered. We've explored this idea and opted to use the plain language search only. I wanted to call this out in case this gets brought back up in Midpoint (whenever we go). 
2. Maybe something the content team can look at, but definitely something we should discuss more within the team. Currently users are able to select "View all health services" and search for everything. 

![Screenshot 2023-08-21 at 4 11 28 PM](https://github.com/department-of-veterans-affairs/va.gov-cms/assets/104526584/8f183b8f-80f0-4524-bcc1-d22df7584f4c)

In my mockups right now, I've changed that to "See results for all VA health services" which is a bit confusing because of the wording and placement. It looks like this is help text for the search bar. 

![Screenshot 2023-08-21 at 4 13 41 PM](https://github.com/department-of-veterans-affairs/va.gov-cms/assets/104526584/3066b9a0-ac9c-4ff2-9c97-3b9383da956d)

This is nothing we need to solve immediately, but I wanted to leave this problem within a ticket so that it's documented. I've also commented within the Sketch file so we don't lose this.
